### DISCUSSION TITLE
Signed PDF from server not LTV enabled

### DISCUSSION THREAD

**Discussion by assidiqi598** [MatthiasValvekens/pyHanko#213] [Q&A] [answered]
Hi Community,

First of all, I thank Matthias for making such a great library. It is really helpful for our project.

I use a certificate from a CA to sign pdf documents. In localhost everything is working fine. But the documents which were signed in our kubernetes server have their LTV not enabled. 

Has anyone had similar problem? Can anyone please tell me where the problem might be?

Here is the code:

```
signer = AsyncAKVSigner(
        key=key,
        credential=credential,
        signing_cert=cert,
        signature_mechanism=algos.SignedDigestAlgorithm(
            {'algorithm': 'sha512_rsa'}),
        signature_mechanism_akv_id=SignatureAlgorithm.rs512)
    
    pdfDoc = args["source"]

    with open(pdfDoc, 'rb+') as inf:
        w = IncrementalPdfFileWriter(inf, strict=False)

        meta = signers.PdfSignatureMetadata(
            field_name='Sig-1',
            embed_validation_info=True,
            validation_context=ValidationContext(
                trust_roots=[
                    SectigoRSADocumentSigningCA,
                    AAACertificateServices,
                    USERTrustRSAAAACA
                ],
                allow_fetching=True),
            use_pades_lta=True,    
        )

        tst_client = timestamps.HTTPTimeStamper(
            'http://timestamp.sectigo.com')

        pdfDoc = pdfDoc[0: len(pdfDoc) - 4] + '-cert.pdf'

        with open(pdfDoc, 'wb') as outf:
            await signers.async_sign_pdf(w,
                                         meta,
                                         signer=signer,
                                         timestamper=tst_client,
                                         output=outf,
                                         existing_fields_only=False)
```

Here is the result:

![image](https://user-images.githubusercontent.com/51270097/216381045-7dea03e8-8ce8-4e50-9d22-269fd8f19fd2.png)

Verification result:

There are also signer info and fingerprints in the actual result.

Integrity
---------
The signature is cryptographically sound.

The digest algorithm used was 'sha512'.
The signature mechanism used was 'sha512_rsa'.

Signing time
------------
Signing time as reported by signer: 2023-02-02T14:48:28+00:00

Signature timestamp token: 2023-02-02T14:48:28+00:00
The token is guaranteed to be newer than the signature.
This timestamp is backed by a time stamping authority.
The timestamp token is cryptographically sound.
TSA certificate subject: "Common Name: Sectigo RSA Time Stamping Signer #3, Organization: Sectigo Limited, State/Province: Manchester, Country: GB"
TSA certificate SHA1 fingerprint: 
TSA certificate SHA256 fingerprint: 
TSA cert trust anchor: "Common Name: USERTrust RSA Certification Authority, Organization: The USERTRUST Network, Locality: Jersey City, State/Province: New Jersey, Country: US"
The TSA certificate is trusted.

Content timestamp token: 2023-02-02T14:48:28+00:00
The token is guaranteed to be older than the signature.
This timestamp is backed by a time stamping authority.
The timestamp token is cryptographically sound.
TSA certificate subject: "Common Name: Sectigo RSA Time Stamping Signer #3, Organization: Sectigo Limited, State/Province: Manchester, Country: GB"
TSA certificate SHA1 fingerprint: 
TSA certificate SHA256 fingerprint: 
TSA cert trust anchor: "Common Name: USERTrust RSA Certification Authority, Organization: The USERTRUST Network, Locality: Jersey City, State/Province: New Jersey, Country: US"
The TSA certificate is trusted.

Modifications
-------------
The signature covers the entire file.

Bottom line
-----------
The signature is judged VALID.

**Comment by assidiqi598** [comment, marked as answer]
Sorry, I should have read the documentation more carefully. I had to set the `subfilter` to pades and use the most relevant certificates for the `trust_roots`

**Reply by MatthiasValvekens** [reply]
Glad to see that you found a solution. If you run into similar issues in the future: the "LTV enabled" status is not actually standardised, and the precise criteria aren't documented anywhere in public view, but it seems to boil down roughly to "Acrobat didn't have to fetch any additional validation data to check the signer's certificate". It's usually possible to track down exactly what's lacking by clicking around the "Signer's Certificate" dialog in the signature settings until you find something that doesn't look right (untrusted cert, missing intermediate certs, validation info missing, etc...). It's not a foolproof method, but it works more often than not in my experience.
